subsys/bluetooth/shell/audio.c

```diff
@@ -815,11 +827,17 @@ static void discover_cb(struct bt_conn *conn, struct bt_codec *codec,
 	}
 
 	if (ep) {
+#if CONFIG_BT_AUDIO_UNICAST_CLIENT_ASE_SNK_COUNT > 0
 		if (params->dir == BT_AUDIO_DIR_SINK) {
 			add_sink(ep, params->num_eps);
-		} else if (params->dir == BT_AUDIO_DIR_SOURCE) {
+		}
+#endif /* CONFIG_BT_AUDIO_UNICAST_CLIENT_ASE_SNK_COUNT > 0 */
+
+#if CONFIG_BT_AUDIO_UNICAST_CLIENT_ASE_SRC_COUNT > 0
+		if (params->dir == BT_AUDIO_DIR_SOURCE) {
 			add_source(ep, params->num_eps);
 		}
+#endif /* CONFIG_BT_AUDIO_UNICAST_CLIENT_ASE_SRC_COUNT > 0*/
 
```

Comment:
Is it possible for apps to reach this point with `dir` set to an invalid value? If so, do you want some kind of assertion here?

Reply:
If so, then we have a problem in the stack. IMO it should not be the upper layers' (e.g. the application) responsibility to check the lower layers (e.g. the stack) correctness :)

Reply:
my concern was the opposite -- is it possible for a mis-configured application to end up in this API with the wrong value somehow. It sounds like the answer is 'no', which addresses my concern.